src/fileimpl.cpp

```diff
@@ -152,6 +162,14 @@ namespace zim
     }
   }
 
+
+  article_index_type FileImpl::getArticleByClusterOrder(article_index_type idx) const
+  {
+      if (idx >= articleListByCluster.size())
+          throw ZimFileFormatError("article index out of range");
```

Comment:
This is not a ZimFileFormatError, the file is correct. This is probably a std::out_of_range.